src/librustc/session/config.rs

```diff
@@ -1932,6 +1932,15 @@ pub fn build_session_options_and_crate_config(
         }
     }
 
+    if cg.no_prepopulate_passes {
+        match cg.lto {
+            Lto::No | Lto::Fat | Lto::Yes => {},
+            Lto::Thin | Lto::ThinLocal =>
+                early_error(error_format, "-Cno-prepopulate-passes is incompatible with thin LTO"),
```

Comment:
How come this does not get triggered when you specify `-Cno-prepopulate-passes` with `-Cpasses=name-anon-globals`?

IMHO this snippet of code should not exist, but I cannot figure out why its presence does not make tests to fail (by erroring earlier than the check in `librustc_codegen_llvm`)!